From 74d22f340fee19138a839624d8bd0e96fc71d998 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthew=20=F0=9F=AA=B7?= <matthew@urgero.net>
Date: Sun, 4 May 2025 01:31:38 -0500
Subject: [PATCH 13/46] Update `vFixedSeeds` with new, reliable nodes (#12)

Update `vFixedSeeds` with new nodes that are known and reliable. There
may be additional updates to this list in the future. For now, 4
reliable nodes is sufficient for bootstrapping new nodes.

Also clean out the old, dead DNS seeder nodes and add the new DNS
endpoint for `seed.lotusia.org`. This DNS endpoint is not active due to
issues with the `lotus-seeder` executable and will need to be addressed
in the future.
---
 contrib/seeds/makeseeds.py   |  2 +-
 contrib/seeds/nodes_main.txt | 11 ++++-------
 src/chainparams.cpp          |  3 +--
 src/chainparamsseeds.h       | 13 +++++--------
 src/seeder/options.cpp       |  1 +
 5 files changed, 12 insertions(+), 18 deletions(-)

diff --git a/contrib/seeds/makeseeds.py b/contrib/seeds/makeseeds.py
index 1d76ab17c..f14d0e337 100755
--- a/contrib/seeds/makeseeds.py
+++ b/contrib/seeds/makeseeds.py
@@ -36,7 +36,7 @@ PATTERN_ONION = re.compile(
 
 # Used to only select nodes with a user agent string compatible with the
 # eCash network.
-PATTERN_AGENT = re.compile(r"^(/lotusd:3.(\d+).(\d+)\(.+\)/)")
+PATTERN_AGENT = re.compile(r"^(/lotusd:8.(\d+).(\d+)\(.+\)/)")
 
 def parseline(line):
     sline = line.split()
diff --git a/contrib/seeds/nodes_main.txt b/contrib/seeds/nodes_main.txt
index a50d801b0..72a4a0a84 100644
--- a/contrib/seeds/nodes_main.txt
+++ b/contrib/seeds/nodes_main.txt
@@ -1,7 +1,4 @@
-35.184.152.63:10605
-45.119.84.253:10605
-65.21.88.223:10605
-116.110.6.191:10605
-123.20.255.166:10605
-152.165.70.246:10605
-173.208.206.181:10605
+147.135.88.232
+147.135.88.233
+82.65.223.81
+80.14.214.145
\ No newline at end of file
diff --git a/src/chainparams.cpp b/src/chainparams.cpp
index ea22c8cc7..d2278fcba 100644
--- a/src/chainparams.cpp
+++ b/src/chainparams.cpp
@@ -166,8 +166,7 @@ public:
         vFixedSeeds.clear();
         vSeeds.clear();
 
-        vSeeds.emplace_back("mainnet.seeds.cash");
-        vSeeds.emplace_back("seed.be.cash");
+        vSeeds.emplace_back("seed.lotusia.org");
 
         base58Prefixes[PUBKEY_ADDRESS] = std::vector<uint8_t>(1, 0);
         base58Prefixes[SCRIPT_ADDRESS] = std::vector<uint8_t>(1, 5);
diff --git a/src/chainparamsseeds.h b/src/chainparamsseeds.h
index d28a4d25a..5edc4b798 100644
--- a/src/chainparamsseeds.h
+++ b/src/chainparamsseeds.h
@@ -8,16 +8,13 @@
  * IPv4 as well as onion addresses are wrapped inside an IPv6 address accordingly.
  */
 static SeedSpec6 pnSeed6_main[] = {
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x23,0xb8,0x98,0x3f}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x2d,0x77,0x54,0xfd}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x41,0x15,0x58,0xdf}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x74,0x6e,0x06,0xbf}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x7b,0x14,0xff,0xa6}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x98,0xa5,0x46,0xf6}, 10605},
-    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0xad,0xd0,0xce,0xb5}, 10605}
+    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x93,0x87,0x58,0xe8}, 10605},
+    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x93,0x87,0x58,0xe9}, 10605},
+    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x52,0x41,0xdf,0x51}, 10605},
+    {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x50,0x0e,0xd6,0x91}, 10605}
 };
 
 static SeedSpec6 pnSeed6_test[] = {
     {{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x23,0xe0,0x6a,0xf0}, 11605}
 };
-#endif // BITCOIN_CHAINPARAMSSEEDS_H
+#endif // BITCOIN_CHAINPARAMSSEEDS_H
\ No newline at end of file
diff --git a/src/seeder/options.cpp b/src/seeder/options.cpp
index 45019639d..d54976f00 100644
--- a/src/seeder/options.cpp
+++ b/src/seeder/options.cpp
@@ -95,6 +95,7 @@ int CDnsSeedOpts::ParseCommandLine(int argc, const char **argv) {
         filter_whitelist.insert(NODE_NETWORK | NODE_BLOOM);
         filter_whitelist.insert(NODE_NETWORK_LIMITED);
         filter_whitelist.insert(NODE_NETWORK_LIMITED | NODE_BLOOM);
+        filter_whitelist.insert(NODE_NETWORK | NODE_BLOOM | NODE_NETWORK_LIMITED);
     }
     return CONTINUE_EXECUTION;
 }
-- 
2.39.5 (Apple Git-154)

